shell/common/pipeline.h

```diff
@@ -144,6 +144,9 @@ class Pipeline : public fml::RefCountedThreadSafe<Pipeline<R>> {
 
     {
       std::scoped_lock lock(queue_mutex_);
+      if (queue_.size() == 0) {
```

Comment:
I wonder why `queue_.size() == 0` doesn't have to be handled previously? It should cause a crash in `queue_.pop_front()` when `queue` is empty?

Reply:
We always had a 1:1 correlation between putting a LayerTree on the queue and taking it off, but I wanted to eliminate the queue and so now the task can be scheduled when there is something in the queue, but by the time it actually runs we might have eliminated something in the queue.

When this fix goes beyond the WIP stage, I'll see if I can throttle the task posting so we only have at most one outstanding task and only when we have a waiting LayerTree to process, but it's a lot easier to just let the posted tasks go away on their own.

Minimally, this might need to change to `if (<nextTreePointer> == null)` when I delete the queue, whether or not I make the task posting more conditional.